backend/program/symlink.py

```diff
@@ -27,44 +28,38 @@ class Symlinker():
         host_path (str): The absolute path of the host mount.
         symlink_path (str): The path where the symlinks will be created.
     """
-    def __init__(self, _):
+    def __init__(self, *_):
         self.key = "symlink"
         self.settings = SymlinkConfig(**settings.get(self.key))
         self.initialized = self.validate()
         if not self.initialized:
-            logger.error("Symlink initialization failed due to invalid configuration.")
-            return
+            raise ValueError("Symlink initialization failed due to invalid configuration.")

```

Comment:
This change needs discussion

Reply:
returning back to parent so that Program handles. If we throw exception from symlink module, does Program throw exception? Hm, needs discussion

Reply:
With the current setup, if the settings fail to initialize we would want to halt the program.  If the failure conditions in settings do not cause the program to halt, inconsistent behavior is to be expected, which can lead the cached pickle data to be in a state that prevents proper functioning when the app is restarted.

Settings should be validated before any actions by the application are taken.